checks/evaluation/pinned_dependencies.go

```diff
@@ -49,6 +51,141 @@ const (
+		checker.DependencyUseTypeNpmCommand:               5,
+		checker.DependencyUseTypePipCommand:               6,
+		checker.DependencyUseTypeNugetCommand:             7,
+	}
+	intToDepType = map[int]checker.DependencyUseType{
+		0: checker.DependencyUseTypeGHAction,
+		1: checker.DependencyUseTypeDockerfileContainerImage,
+		2: checker.DependencyUseTypeDownloadThenRun,
+		3: checker.DependencyUseTypeGoCommand,
+		4: checker.DependencyUseTypeChocoCommand,
+		5: checker.DependencyUseTypeNpmCommand,
+		6: checker.DependencyUseTypePipCommand,
+		7: checker.DependencyUseTypeNugetCommand,
+	}
+)
+
+func ruleRemToProbeRem(rem *rule.Remediation) *probe.Remediation {
+	return &probe.Remediation{
+		Patch:    rem.Patch,
+		Text:     rem.Text,
+		Markdown: rem.Markdown,
+		Effort:   probe.RemediationEffort(rem.Effort),
+	}
+}
+
+func probeRemToRuleRem(rem *probe.Remediation) *rule.Remediation {
+	return &rule.Remediation{
+		Patch:    rem.Patch,
+		Text:     rem.Text,
+		Markdown: rem.Markdown,
+		Effort:   rule.RemediationEffort(rem.Effort),
+	}
+}
+
+func dependenciesToFindings(deps []checker.Dependency) ([]finding.Finding, error) {
+	findings := make([]finding.Finding, 0)
+	for i := range deps {
+		rr := deps[i]
+		if rr.Location == nil {
+			if rr.Msg == nil {
+				e := sce.WithMessage(sce.ErrScorecardInternal, "empty File field")
+				return findings, e
+			}
+			f := &finding.Finding{
+				Probe:   "",
+				Outcome: finding.OutcomeNotApplicable,
+				Message: *rr.Msg,
+			}
+			findings = append(findings, *f)
+			continue
+		}
+		if rr.Msg != nil {
+			loc := &finding.Location{
+				Type:      rr.Location.Type,
+				Path:      rr.Location.Path,
+				LineStart: &rr.Location.Offset,
+				LineEnd:   &rr.Location.EndOffset,
+				Snippet:   &rr.Location.Snippet,
+			}
+			f := &finding.Finding{
+				Probe:    "",
+				Outcome:  finding.OutcomeNotApplicable,
+				Message:  *rr.Msg,
+				Location: loc,
+			}
+			findings = append(findings, *f)
+			continue
+		}
+		if rr.Pinned == nil {
+			loc := &finding.Location{
+				Type:      rr.Location.Type,
+				Path:      rr.Location.Path,
+				LineStart: &rr.Location.Offset,
+				LineEnd:   &rr.Location.EndOffset,
+				Snippet:   &rr.Location.Snippet,
+			}
+			f := &finding.Finding{
+				Probe:    "",
+				Outcome:  finding.OutcomeNotApplicable,
+				Message:  fmt.Sprintf("%s has empty Pinned field", rr.Type),
+				Location: loc,
+			}
+			findings = append(findings, *f)
+			continue
+		}
+		if !*rr.Pinned {
+			loc := &finding.Location{
+				Type:      rr.Location.Type,
+				Path:      rr.Location.Path,
+				LineStart: &rr.Location.Offset,
+				LineEnd:   &rr.Location.EndOffset,
+				Snippet:   &rr.Location.Snippet,
+			}
+			f := &finding.Finding{
+				Probe:    "",
+				Outcome:  finding.OutcomeNegative,
+				Message:  generateTextUnpinned(&rr),
+				Location: loc,
+			}
+			if rr.Remediation != nil {
+				f.Remediation = ruleRemToProbeRem(rr.Remediation)
+			}
+			f = f.WithValues(map[string]int{
```

Comment:
I think we might want to have a distinct probes for each ecosystem, because remediation steps will be different. Not a blocker for this PR since it's an intermediary PR. Unless you think it'd help to do it now?

Reply:
Good to know, thanks! I think it is best to introduce the probes later to avoid a large PR.

Reply:
Yes, it would be temporary, but perhaps we should only have either probe.Remediation or rule.Remediation in Scorecard, since they are identical?

Reply:
as far as I know, probes and rules are the same thing. It just got renamed. Laurent would know for sure

Reply:
I confirm. rule is the old name we used.

Reply:
> but perhaps we should only have either probe.Remediation or rule.Remediation in Scorecard, since they are identical?

If this is just a code smell thing, and not blocking anything, I'm still in favor of this being done later, tracked in an issue to be taken care of closer to the major version bump

Reply:
Sounds good. I have created https://github.com/ossf/scorecard/issues/3677 to track this. Closing this issue.